tests/unit/woopay/test-class-woopay-utilities.php

```diff
@@ -191,6 +191,62 @@ public function test_should_save_platform_customer_in_classic_checkout() {
 		unset( $_POST['save_user_in_woopay'] );
 	}
 
+	/**
+	 * WooPay should be enabled for guest checkout when user is logged in.
+	 *
+	 * @return void
+	 */
+	public function test_should_enable_woopay_on_guest_checkout_logged_in() {
+		wp_set_current_user( 1 );
+
+		$woopay_utilities = new WooPay_Utilities();
+
+		$this->assertTrue( $woopay_utilities->should_enable_woopay_on_guest_checkout() );
+	}
+
+	/**
+	 * WooPay should be enabled for guest checkout when user is not logged in and guest checkout is enabled.
+	 *
+	 * @return void
+	 */
+	public function test_should_enable_woopay_on_guest_checkout_logged_out_guest_enabled() {
+		wp_set_current_user( 0 );
+		update_option( 'woocommerce_enable_guest_checkout', 'yes' );
```

Comment:
Should we reset this option before running other tests?

Reply:
Fixed on https://github.com/Automattic/woocommerce-payments/pull/10917/commits/c0558a5d8d78385ae391b252e9288b47b1b866af.